### ISSUE TITLE
Unable to use along with react-native-firebase due to issue with `use_frameworks!`

### ISSUE THREAD

**Issue by sanjaypojo** [voximplant/react-native-voximplant#68]
Hi @YuliaGrigorieva,

Is there any way to use Voximplant without `use_frameworks`? I'm unable to use `react-native-firebase` and a couple of other libraries due to an issue where these libraries just don't work whenever `use_frameworks` seems to be used. Alternately, is there a way we could make it work by using a bridging header for Voximplant as explained in [this issue](https://github.com/ivpusic/react-native-image-crop-picker/issues/663)?

Any workaround to use your library along with react-native-firebase would be amazing!

**Comment by sanjaypojo** [comment]
[Here's another link to someone suggesting its possible to avoid using use_frameworks!](https://stackoverflow.com/a/49469205/4239307)

**Comment by sanjaypojo** [comment]
And a third option I have seen is to use `modular_headers` instead of `use_frameworks`

**Comment by YuliaGrigorieva** [comment]
Hello!

`use_frameworks!` is required not because we use swift in the Voximplant iOS SDK or RN native iOS module, but due to the Voximplant iOS SDK dependencies. Voximplant iOS SDK and RN iOS is written only on objective-c.

`use_frameworks!` means that the pod dependencies should be compiled as frameworks. If this option is not indicated, they will be compiled as static libs. 

Voximplant iOS SDK has CocoaLumberjack and SocketRocket dependencies that are dynamically linked to the Voximplant iOS SDK. Building them as static libs will cause the crash on the application launch, i.e. Voximplant iOS SDK won't be able to find its dependencies runtime.

Unfortunately, according to this the options that you propose will not resolve the issue. 
We are currently investigating possible workaround for now to resolve your issue and looking for a long-term solution.

Best regards,
Yulia Grigorieva

**Comment by sanjaypojo** [comment]
Hi Yulia, thanks a lot for the quick response. I understand the issue -- it would be great if you could provide a workaround as we need to roll out push notifications soon and we're currently blocked by this problem. Thanks a lot!

And a long term solution would be really awesome.

**Comment by sanjaypojo** [comment]
Do you think there’s a fix possible for `react-native-firebase`?

If it’s not possible or will take time, would it be better to use `react-native-notifications` like you have used in the demo repo?

**Comment by sanjaypojo** [comment]
Hi @YuliaGrigorieva, we are thinking of going ahead and using `react-native-notifications`. Please let us know if you think there are drawbacks with this approach as you have used this before. Thanks a lot!

**Comment by YuliaGrigorieva** [comment]
Hello @sanjaypojo!

Yes, I think the best option for now is to use `react-native-notifications`. 

Unfortunately, I can't provide any suitable workaround for `react-native-firebase` now. However we are considering this issue and will develop the solution for it.

Best regards,
Yulia Grigorieva

**Comment by sanjaypojo** [comment]
Sounds good, thanks @YuliaGrigorieva!

**Comment by YuliaGrigorieva** [comment]
@sanjaypojo,

One more thing that I have missed to point.
RNFirebase as well as Firebase iOS SDK does not support VoIP push notifications that are required to receive incoming calls (PushKit)
Please check this issue: https://github.com/invertase/react-native-firebase/issues/987

However we are working on a solution for RNFirebase integration conflicts and will update the issue as soon as it is available.

Best regards,
Yulia Grigorieva

**Comment by tmjordan** [comment]
@sanjaypojo @YuliaGrigorieva i think we must find a solution to this, because it's not only rnfirebase which has coflicts with use_frameworks. I have a lot of libraries which don't work with that, and it is a problem. So i can't use voximplant in my project. 
Is there a way to install voximplant without using cocoapods? It would be nice to have another option.
But anyway thanks for that amazing library.

**Comment by YuliaGrigorieva** [comment]
Hello, @tmjordan !

We will remove `use_frameworks!` requirement for iOS installation in the next Voximplant React Native SDK release. However we do not plan to support non-cocoapods installation in the near future.

I will update the issue once we publish a new release.

Best regards,
Yulia Grigorieva

**Comment by tmjordan** [comment]
Hi @YuliaGrigorieva, that's a great new. Thanks. I really want to use voximplant.
Can we have an idea of when the new release will be available?

**Comment by YuliaGrigorieva** [comment]
Hello, @tmjordan !

We plan it for the next week.

Best regards,
Yulia Grigorieva

**Comment by tmjordan** [comment]
Thanks @YuliaGrigorieva

**Comment by YuliaGrigorieva** [comment]
Hello!

Today we have released Voximplant React Native SDK 1.6.0. Starting this release `use_frameworks!` is no longer mandatory.

Best regards,
Yulia Grigorieva
